src/java_bytecode/library/Makefile

```diff
@@ -1,3 +1,46 @@
-all:
-	${MAKE} -C .. core-models.jar
-	cp -f ../core-models.jar ./
+include ../../config.inc
+include ../../common
+
+SOURCE_DIR := src
+BINARY_DIR := classes
+
+FIND := find
+
+JAVAC := javac
+JFLAGS := -sourcepath $(SOURCE_DIR) -d $(BINARY_DIR) -XDignore.symbol.file
+
+CLASSPATH := SOURCE_DIR
+
+ALL_JAVAS := $(wildcard $(SOURCE_DIR)/*/*.java $(SOURCE_DIR)/*/*/*.java $(SOURCE_DIR)/*/*/*/*.java) #$(shell find $(SOURCE_DIR) -name '*.java')
+ALL_CLASSES := $(patsubst $(SOURCE_DIR)/%.java,$(BINARY_DIR)/%.class,$(ALL_JAVAS))
+
+$(BINARY_DIR):
+	mkdir -p $(BINARY_DIR)
+
+.SUFFIXES: .java .class
+
+$(BINARY_DIR)/%.class : $(SOURCE_DIR)/%.java $(BINARY_DIR)
+	$(JAVAC) $(JFLAGS) $(patsubst $(BINARY_DIR)/%.class,$(SOURCE_DIR)/%.java,$@)
+
+JAR := jar
+JARFLAGS := -cf
+
+.NOTPARALLEL: core-models.jar
```

Comment:
Remove the dependencies of the `.NOTPARALLEL` target and move it to the top of the file. Add a comment explaining that we use it to prevent parallel compilation of the java files and why.

Reply:
done